elftools/dwarf/die.py

```diff
@@ -142,6 +167,40 @@ def set_parent(self, die):
 
     #------ PRIVATE ------#
 
+    def _search_for_ancestors(self):
+        """ Search for our parent by, starting with the CU top die,
+            setting the _parent link of each child and then iterating on
+            the child which was closest to object DIE offset without over,
+            whithan to the object DIE offset, which is an ancestor.
+        """
+        search = self.cu.get_top_DIE()
+
+        while search.offset < self.offset:
+
+            prev = search
+            for child in search.iter_children():
+                child.set_parent(search)
+                if child.offset <= self.offset:
+                    prev = child
+
+            # We need to check if the offset is the terminator
+            if search.has_children and search._terminator.offset <= self.offset:
+                    prev = search._terminator
+
+            # If we didn't find a closer parent, give up, don't loop.
+            # Either we mis-parsed an ancestor or someone asked for a DIE
+            # by an offset that was not actually the start of a DIE.
+            if prev is search:
+                raise ValueError("offset %s not in CU %s DIE tree" %
+                    (self.offset, self.cu.cu_offset))
+
+            search = prev
+
+        pass
```

Comment:
Not sure what's going on here. `pass` outside the loop? 

Should this line be removed?

Also the commented code following it

Reply:
refactoring remenants, fixed.